dot/core/syncer.go

```diff
@@ -125,19 +125,28 @@ func NewSyncer(cfg *SyncerConfig) (*Syncer, error) {
 
 // Start begins the syncer
 func (s *Syncer) Start() {
+	// thread safe change stopped to 1
+	canLock := atomic.CompareAndSwapUint32(&s.stopped, 0, 1)
+	if !canLock {
+		panic("[sync] Error when trying to change Syncer from stopped to started.")
+	}
 	go s.watchForBlocks()
 	go s.watchForResponses()
 }
 
 // Stop stops the syncer
 func (s *Syncer) Stop() {
 	// stop goroutines
-	s.stopped = true
+	canUnlock := atomic.CompareAndSwapUint32(&s.stopped, 1, 0)
+	if !canUnlock {
+		panic("[sync] Error when trying to change Syncer from started to stopped.")
```

Comment:
return error

Reply:
The stop method does not return error right now, this will add extra logic into this PR. Are you sure ?

Reply:
yes, the syncer is only stopped in one place, so I don't think it's a lot of logic to add for an err check.

Reply:
also, you are changing `s.stopped` to 0 in the Stop method, shouldn't it be changed to `1`?

Reply:
I updated this to started and updated the method signature as requested, please re-check.